### PR TITLE
Sales commission so based module

### PR DESCRIPTION
**Pull request by maksymkv25**

### CODE REVIEW

**Comment by maksymkv25** [comment]
@pedrobaeza Hi, the module is ready. Do I need to do something extra?

**Comment by pedrobaeza** [comment]
Formally, you can squash commits together, as they don't add value being unfolded. On the contributing process, at least 2 persons must review the PR, one of them being PSC:

https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#review

Unfortunately, I can't physically review every PR that enters OCA. You can enroll people to review your PR reviewing other existing PRs and asking in exchange that they review yours.

**Comment by maksymkv25** [comment]
@dcorio @joao-p-marques 
Hello, can you please review this PR? Thank you

**Comment by francesco-ooops** [comment]
@maksymkv25 you should make sure PR is compliant with https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst before asking for reviews

Also there is no documentation for this module, so functionals can't provide their opinion on whether it works or not

**Comment by jguenat** [comment]
Hello @maksymkv25 ,
Thanks for your PR, I tested the module quickly and it looks promising.

Concerning documentation you should create a readme folder containing .rst files (USAGE.rst etc...). Then the oca bot will create the main README.rst automatically after merge  (you can do it manually if you want https://github.com/OCA/maintainer-tools#readme-generator).

File: sale_commission_so_based/models/commission_settlement.py
Comment (jguenat) [inline sale_commission_so_based/models/commission_settlement.py]:
```suggestion
        string="Source sale line",
```

**Review by jguenat — COMMENTED** [review]
Also it would be grate to have the "Source sale line" on the commission.settlement view & report, same as what is done with the Source invoice line in account_commission.

File: sale_commission_so_based/readme/DESCRIPTION.rst
Comment (francesco-ooops) [inline sale_commission_so_based/readme/DESCRIPTION.rst]: question: how is module sale_commission_salesman related to this module? I don't see it as a dependency

Comment (maksymkv25) [reply sale_commission_so_based/readme/DESCRIPTION.rst]: @francesco-ooops You are right, typo in the description. Thank's

File: sale_commission_so_based/models/commission_settlement.py
Comment (maksymkv25) [reply sale_commission_so_based/models/commission_settlement.py]: @jguenat I've just added source sale line on view. Thanks

**Review by francesco-ooops — APPROVED** [review]
Functional ok

squash commits please

**Comment by maksymkv25** [comment]
This PR will be closed, because had a problem whit heads to squash commits. 
New [PR](https://github.com/OCA/commission/pull/473)

**Pull request closed without merging**
